plugins/gimp/file-jxl-save.cc

```diff
@@ -3,130 +3,813 @@
+  gtk_frame_set_shadow_type(GTK_FRAME(frame), GTK_SHADOW_NONE);
+  gtk_box_pack_start(GTK_BOX(vbox), frame, false, false, 0);
+  gtk_widget_show(frame);
+
+  // Faster Decoding Scale
+  table = gtk_table_new(1, 3, false);
+  gtk_table_set_col_spacings(GTK_TABLE(table), 6);
+  gtk_container_add(GTK_CONTAINER(frame), table);
+  gtk_widget_show(table);
+
+  entry_faster = (GtkAdjustment*)gimp_scale_entry_new(
+      GTK_TABLE(table), 0, 0, "", SCALE_WIDTH, 0, jxl_save_opts.faster_decoding,
+      0, 5, 1, 1, 0, true, 0.0, 0.0, faster_help, "file-jxl-save");
+
+  advanced_opts.push_back((GtkWidget*)entry_faster);
+
+  // Faster Decoding Signals
+  g_signal_connect(entry_faster, "value-changed",
+                   G_CALLBACK(gimp_int_adjustment_update),
+                   &jxl_save_opts.faster_decoding);
+
+  // Enable Advanced Settings
+  frame = gtk_frame_new(0);
+  gimp_help_set_help_data(frame, "Advanced Settings shouldn't be used.",
+                          nullptr);
+  gtk_frame_set_shadow_type(GTK_FRAME(frame), GTK_SHADOW_NONE);
+  gtk_box_pack_start(GTK_BOX(main_vbox), frame, true, true, 0);
+  gtk_widget_show(frame);
+
+  vbox = gtk_vbox_new(false, 6);
+  gtk_container_set_border_width(GTK_CONTAINER(vbox), 6);
+  gtk_container_add(GTK_CONTAINER(frame), vbox);
+  gtk_widget_show(vbox);
+
+  static gchar advanced_help[] =
+      "Advanced Settings shouldn't be used.  "
+      "Check this box if you want to use them anyway.";
+  toggle = gtk_check_button_new_with_label("Enable Advanced Settings");
+  gimp_help_set_help_data(toggle, advanced_help, nullptr);
+  gtk_toggle_button_set_active(GTK_TOGGLE_BUTTON(toggle),
+                               jxl_save_opts.advanced_mode);
+  gtk_box_pack_start(GTK_BOX(vbox), toggle, false, false, 0);
+  gtk_widget_show(toggle);
+
+  g_signal_connect(toggle, "toggled", G_CALLBACK(gui_on_change_advanced_mode),
+                   &advanced_opts);
+
+  // show dialog
+  gtk_widget_show(dialog);
+
+  GtkAllocation allocation;
+  gtk_widget_get_allocation(dialog, &allocation);
+
+  int height = allocation.height;
+  gtk_widget_set_size_request(dialog, height * 1.4, height);
+
+  run = (gimp_dialog_run(GIMP_DIALOG(dialog)) == GTK_RESPONSE_OK);
+  gtk_widget_destroy(dialog);
+
+  return run;
+}  // save_dialog
+
+JpegXlSaveOpts::JpegXlSaveOpts() {
+  set_distance(1.0f);
+
+  advanced_mode = false;
+  lossless = false;
+  is_linear = false;
+  has_alpha = false;
+  save_exif = false;
+  use_container = true;
+
+  faster_decoding = 0;
+  encoding_effort = 7;
+
+  pixel_format.num_channels = 4;
+  pixel_format.data_type = JXL_TYPE_UINT8;
+  pixel_format.endianness = JXL_NATIVE_ENDIAN;
+  pixel_format.align = 0;
+
+  basic_info.alpha_bits = 0;
+  basic_info.alpha_exponent_bits = 0;
+  basic_info.uses_original_profile = false;
+  basic_info.intensity_target = 255.0f;
+  basic_info.orientation = JXL_ORIENT_IDENTITY;
+
+  babl_format_str = "RGB u16";
+  babl_type_str = "u16";
+  babl_model_str = "RGB";
```

Comment:
FYI, in modern C++, it is also possible to initialize members in the class declaration directly. (But up to you.)

https://en.cppreference.com/w/cpp/language/data_members#Member_initialization (2)

Reply:
So this shows how long it's been since I've done anything with C++...  I'd personally rather initialize them directly.  Not sure if I should do that for this commit or at a later time, since copy/paste can lead to errors.

Reply:
I think what Sami meant is that you can do 

```
JpegXlSaveOpts::JpegXlSaveOpts(): advanced_mode(false),  lossless(false), ... {
  set_distance(1.0f);
}
```

which I personally would find more readable - but I'm not sure and I don't really have that strong of an opinion :P

Reply:
I was thinking of even putting them outside of the constructor, directly in the class declaration:

```c++
class JpegXlSaveOpts {
 public:
  bool advanced_mode = false;
  bool lossless = false;
  // …
};
```